src/lib/scheduler/node_queue_scheduler.hpp

```diff
@@ -12,63 +12,54 @@ namespace hyrise {
- * For setting up a Scheduler a topology is used. A topology encapsulates the machine's architecture, e.g. number
- * of CPUs and the number of nodes, where a node is a cluster of CPUs.
- * In general, each node owns a TaskQueue. Furthermore, one Worker is assigned to one CPU. Therefore, the Worker
- * running on CPUs of one node are just pulling from the single TaskQueue of this node.
+ * The Scheduler is the main entry point and (currently) there are the ImmediateExecutionScheduler (single-threaded) 
+ * and the NodeQueueScheduler (multi-threaded). For setting up the NodeQueueScheduler the server's topology is used. A
+ * topology encapsulates the machine's architecture, e.g., the number of CPU threads and NUMA nodes, where a node is
+ * typically a socket or CPU (usually having multiple threads/cores).
+ * Each node owns a TaskQueue. Furthermore, one Worker is assigned to one CPU thread. The Worker running on one CPU
+ * thread of a node is primarily pulling from the local TaskQueue of this node.
  *
- * A topology can also be created with Hyrise::get().topology.use_fake_numa_topology() to simulate a NUMA system
- * with multiple nodes (queues) and worker and should mainly be used for testing NUMA-concepts
- * on non-NUMA development machines.
+ * A topology can also be created with Hyrise::get().topology.use_fake_numa_topology() to simulate a NUMA system with
+ * multiple nodes (thus, queues) and Workers and should mainly be used for testing NUMA-concepts on non-NUMA
+ * development machines.
  *
  *
  * WORK STEALING
  *
- * Currently, a simple work stealing is implemented. Work stealing is useful to avoid idle workers (and therefore
- * idle CPUs) while there are still tasks in the system that need to be processed. A worker gets idle if it can not
+ * Currently, a simple work stealing is implemented. Work stealing is useful to avoid idle Workers (and therefore idle
+ * CPU threads) while there are still tasks in the system that need to be processed. A Worker gets idle if it can not
  * pull a ready task. This occurs in two cases:
  *  1) all tasks in the queue are not ready
  *  2) the queue is empty
- * In both cases, the current worker is checking another queue for a ready task. Checking another queue means accessing
- * another node (remote node). As of the physical distance of nodes, accessing a remote node is ~1.6 times slower than
- * accessing a local node. [1]
- * Therefore, the current worker fetches a task from a remote queue (without removing it). Then, the current worker
- * is sleeping some milliseconds to give any local worker of the remote node the chance to pull that task. If no local
- * worker of the remote node pulled the task, the current worker is pulling the task and therefore steals it.
- * Afterwards, the current worker is checking its local queue gain.
+ * In both cases, the current Worker is checking non-local queues of other NUMA nodes for ready tasks. The Worker pulls
```

Comment:
queues, task lowercase again (and following). `Ready tasks` does not sound correct here as well.

Reply:
I agree with `TaskQueues`, but when talking in general about a task or a scheduler (and not about a concrete implementation like the `JobTask` or the `NodeQueueScheduler`), I would leave it as `task` and `scheduler`.

What's your view on that?

Reply:
Totally fine with task, scheduler (and also prefer table, chunk, segment etc.). Just want it to be consistent.